Arduino-Bling/README.md

```diff
@@ -0,0 +1,76 @@
+# Integrating NeoPixel with Arduino & Robot Code
+This directory hosts the source code to interface roborio/wpilib robot fuctionality with
+NeoPixel/Arduino.
+
+## High-level code flow
+1. Robot code sets the desired LED blink pattern
+2. LED subsystem sends the required blink pattern over the serial port
+   to the Arduino
+3. Arduino code frequently checks the serial port to determine if a new
+   blink pattern is required --> if so, it changes the blink pattern
+
+## How to interface robot code w/ Arduino for LED animation
+Robot code interfaces to Arduino via serial port communication using
+UART protocol, implemented as a CDC/ACM class communication device over
+USB. While not "classic" UART, it operates the same. Serial object in
+this case only refers to UART communications. The commands to change
+LED animation pattern is sent as an octet byte over the serial.
+
+Robot code:
+* `LED` subsystem:
+  * specifies the blink patterns --> see `LED::BlinkState`
+  * methods to send the blink code over the serial port --> see
+    `LED::setBlinkState()`. Note, this method is also used to set blink
+    state changes when `SetBlinkStateCommand` cannot be used.
+* `SetBlinkStateCommand` Command for incorporating blink state change
+  requests to robot code.
+
+### Specifying Bling States
+Bling states indicates different bling annimation patterns. As a general
+rule, for easy maintenance, states should use *names* that reflect robot behavior over
+bling patterns.
+
+```
+  public enum BlingState {
+        OFF("0"),           // turn off the LED animation -- not used in robot code
```

Comment:
Delete the ' -- not used in robot code' because that comment will be out of date before you know it.